designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -115,16 +119,52 @@ object ScenarioStatistics {
     }
   }
 
-  def getComponentStatistic(componentList: List[component.ComponentListElement]): Map[String, String] = {
+  // TODO: Should not depend on DTO, need to extract usageCount and check if all available components are present using processingTypeDataProvider
+  def getComponentStatistic(
+      componentList: List[component.ComponentListElement],
+      components: List[ComponentDefinitionWithImplementation]
+  ): Map[String, String] = {
     if (componentList.isEmpty) {
       Map.empty
     } else {
-      val withoutFragments      = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
-      val componentsByNameCount = withoutFragments.groupBy(_.name).size
 
-      Map(
-        ComponentsCount -> componentsByNameCount
-      ).map { case (k, v) => (k.toString, v.toString) }
+      // Get number of available components to check how many custom components created
+      val withoutFragments = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
```

Comment:
We could add `getAlignedComponentsWithBuiltInComponents` next to the `getAlignedComponentsWithBuiltInComponentsAndFragments`. Thanks to that we don't need to filter out fragments here. Now we add some filter and then we filter out them, it looks strange.

Reply:
This problem will be resolved once we completely switch from using `ComponentService` and that will be when we'll have other way to get `usageCount` for components